src/vs/workbench/parts/terminal/common/terminal.ts

```diff
@@ -17,6 +17,9 @@ export const TERMINAL_SERVICE_ID = 'terminalService';
 
 export const TERMINAL_DEFAULT_RIGHT_CLICK_COPY_PASTE = platform.isWindows;
 
+/**  A context key that is set when the integrated terminal OR the terminal FindWidget has focus. */
+export const KEYBINDING_CONTEXT_TERMINAL_COMPONENTS_FOCUS = new RawContextKey<boolean>('terminalComponentsFocus', undefined);
```

Comment:
Since we already have `KEYBINDING_CONTEXT_TERMINAL_FIND_WIDGET_VISIBLE` can we do an && condition within the `when` clause with `KEYBINDING_CONTEXT_TERMINAL_FOCUS`?

Reply:
@Tyriar 
Hey no problem, I feel bad I have not had the time to help with terminal issues!  Lots of my time
creating and fixing the new drop-down ;-)

Since Find Next/Previous have to work both with the Find wgt focused or the terminal focused
we cannot use && what we really want is not exist - OR

I created the new context key and logic so we had a condition/key that was true
when the terminal is active or the find widget is active.

Reply:
Clarification:

The Commands have to work in three cases:
- terminal active, widget not visible
- terminal active, widget visible, not focused
- terminal active, widget visible, any part of it focused

Reply:
@alexandrudima the proposed solution is to add another context key related to terminal focus to get F3/shift+F3 working with terminal find. Currently there is `terminalFocus` and `terminalFindWidgetInputFocused`, the proposed solution is to add yet another context key to cover the case where either of these are focused (plus any button in the find widget). Is there a better way to do this apart from just adding more and more context keys?

Reply:
@Tyriar There is no support for OR (||) in keybindings `when` clause at this time. The reason for that is that introducing an OR expression would require us implementing an SMT solver to do inverse keybindings lookups with the same quality we do today. See https://github.com/Microsoft/vscode/issues/33547#issuecomment-326213490 for a detailed explanation.

So if you want to write an expression that use `a||b`, there are two possibilities as of today:
1. add a new context key `c` and make sure it is true when `a||b`.
2. add two keybindings, once for `a`, and once for `b`.